test/testvm.py

```diff
@@ -935,25 +1058,31 @@ def rem_disk(self, index):
         if "proc" in disk and disk["proc"] and disk["proc"].poll() == None:
             disk["proc"].terminate()
 
-    def add_netiface(self, mac, vlan=0):
-        if len(mac) == 2:
-            tree = etree.parse(open("./guest/network-cockpit.xml"))
-            for h in tree.find(".//dhcp"):
-                macaddr = h.get("mac")
-                if macaddr:
-                    mac = macaddr[:-2] + mac
-                    break
-        self._monitor_qemu("device_add e1000,vlan=%s,mac=%s" % (vlan,mac));
-        return mac
+    def add_netiface(self, mac=None, secondary_network=False):
+        desc_template = ""
+        with open("./files/test_domain_network.xml", "r") as desc_file:
```

Comment:
I pushed some code to stefwalter/test_libvirt to do this without secondary networks in a more flexible way.